examples/dao/test/failing-path.js

```diff
@@ -0,0 +1,585 @@
+        `int:${votingEnd}`, // votingEnd (now + 3min)
+        `int:${executeBefore}`, // executeBefore (now + 7min)
+        `int:${ProposalType.ALGO_TRANSFER}`, // type
+        `addr:${ctx.daoFundLsig.address()}`, // from (DAO treasury)
+        `addr:${ctx.proposer.address}`, // recepient
+        `int:${2e6}` // amount (in microalgos)
+      ];
+
+      addProposalTx = mkProposalTx(
+        ctx.daoAppID,
+        ctx.govTokenID,
+        ctx.proposer.account,
+        ctx.depositLsig,
+        ctx.proposalLsig,
+        ctx.daoFundLsig
+      );
+      addProposalTx[1].amount = 15;
+    });
+
+    it('should fail if proposalLsig not opted-in to DAO App', () => {
+      assert.throws(() => ctx.executeTx(addProposalTx), APP_NOT_FOUND);
+    });
+
+    it('should fail if votingEnd < votingStart', () => {
+      ctx.optInToDAOApp(ctx.proposalLsig.address()); // opt-in
+      ctx.syncAccounts();
+
+      // set votingEnd < votingStart (30s before)
+      proposalParams[6] = `int:${votingStart - 30}`;
+      addProposalTx[0].appArgs = proposalParams;
+
+      assert.throws(() => ctx.executeTx(addProposalTx), 'RUNTIME_ERR1005: Result of current operation caused integer underflow');
+    });
+
+    it('should fail if "min_duration <= votingEnd - votingStart <= max_duration" is not satisfied', () => {
+      // increase votingEnd (voting duration set as 9min, but allowed is 5min)
+      proposalParams[6] = `int:${votingEnd + (10 * 60)}`;
+      addProposalTx[0].appArgs = proposalParams;
+
+      assert.throws(() => ctx.executeTx(addProposalTx), RUNTIME_ERR1009);
+    });
+
+    it('should fail if executeBefore < votingEnd', () => {
+      // set executeBefore
+      proposalParams[7] = `int:${votingEnd - 30}`;
+      addProposalTx[0].appArgs = proposalParams;
+      assert.throws(() => ctx.executeTx(addProposalTx), RUNTIME_ERR1009);
+    });
+
+    it('should fail if proposal type is not between [1-3]', () => {
+      proposalParams[8] = 'int:4';
+      addProposalTx[0].appArgs = proposalParams;
+      assert.throws(() => ctx.executeTx(addProposalTx), RUNTIME_ERR1009);
+    });
+
+    it('should fail if gov tokens deposit transaction is not present', () => {
+      assert.throws(() => ctx.executeTx({ ...addProposalTx[0] }), INDEX_OUT_OF_BOUND_ERR);
+    });
+
+    it('should fail if gov tokens deposit is not equal to DAO.deposit', () => {
+      addProposalTx[1].amount = 10;
+      assert.throws(() => ctx.executeTx(addProposalTx), RUNTIME_ERR1009);
+    });
+
+    it('should fail if deposit lsig address is different', () => {
+      addProposalTx[1].toAccountAddr = ctx.proposer.address;
+      assert.throws(() => ctx.executeTx(addProposalTx), RUNTIME_ERR1009);
+    });
+
+    it('should fail if votingStart < now', () => {
+      ctx.runtime.setRoundAndTimestamp(1, now + (2 * 60)); // latest time set as (now + 2min)
+      assert.throws(() => ctx.executeTx(addProposalTx), RUNTIME_ERR1009);
+    });
```

Comment:
it should also fail when `votingStart == now`

Reply:
done